### ISSUE TITLE
ssh_citrix_adc connection plugin issue

### ISSUE THREAD

**Issue by mschirrmeister** [citrix/citrix-adc-ansible-modules#203]
The `ssh_citrix_adc` connection plugin does not work with ansible **4.6.0**. The error is about missing settings. There are 3 settings it complains about.

- timeout
- reconnection_retries
- ssh_transfer_method

Here is an example error.
```
$ ansible-playbook -i inventory/development play_foo.yml

PLAY [NetScaler commands] ***

TASK [show uptime] ***
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: KeyError: 'Requested entry (plugin_type: connection plugin: ssh_citrix_adc setting: timeout ) was not defined in configuration.'
fatal: [msvpx1]: FAILED! => {"msg": "Unexpected failure during module execution.", "stdout": ""}
```

My ansible version
```
$ ansible-playbook --version

ansible-playbook [core 2.11.5]
  config file = /Users/marco/MyData/git/citrix-ansible/ms-ansible-netscaler/citrixtest/ansible.cfg
  configured module search path = ['/Users/marco/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/4.6.0/libexec/lib/python3.9/site-packages/ansible
  ansible collection location = /Users/marco/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible-playbook
  python version = 3.9.7 (default, Sep  3 2021, 12:37:55) [Clang 12.0.5 (clang-1205.0.22.9)]
  jinja version = 3.0.1
  libyaml = True
```

My `ansible.cfg`
```
$ cat ansible.cfg | grep -vE '(^#|^$)'

[defaults]
roles_path=roles/internal:roles/external
host_key_checking = False
timeout = 20
local_tmp = ~/.ansible/tmp
retry_files_enabled = False
connection_plugins = ~/.ansible/plugins/connection_plugins/
[ssh_connection]
control_path=%(directory)s/%%h-%%r
scp_if_ssh = True
timeout = 10
```

The error shows up even when I had the timeout configured in **ansible.cfg**, as an **env variable** or as an **argument** on the cli.

As a workaround I defined the options in the `ssh_citrix_adc.py` script and that way the execution is successful.

```
$ diff -Naur ssh_citrix_adc.py.orig ssh_citrix_adc.py.new

--- ssh_citrix_adc.py.orig	2021-09-29 14:30:54.000000000 +0200
+++ ssh_citrix_adc.py.new	2021-09-29 14:06:16.000000000 +0200
@@ -366,4 +366,7 @@
     def exec_command(self, cmd, in_data=None, sudoable=True):
         ''' run a command on the remote host '''

+        self.set_option('timeout', 5)
+        self.set_option('reconnection_retries', 0)
+        self.set_option('ssh_transfer_method', 'scp')
         return super(Connection, self).exec_command(cmd, in_data=in_data, sudoable=sudoable)
```

**Comment by mschirrmeister** [comment]
I just noticed there is already a pull request for this exact problem #197.

**Comment by giorgos-nikolopoulos** [comment]
Fixed with #204
